### PR TITLE
Adding CompositeDependencyResolver 

### PR DESCRIPTION
**Pull request by fhsilva**
This is a new pull request for previous pull 1575.

Robolectric has currently two variables controlling offline behavior:
<b>robolectric.offline</b> and <b>robolectric.dependency.dir</b>
Both are tightly coupled, in the sense that one does not work without the other. They also do not solve all problems related with offline behavior. 
For instance, I want Robolectric to use some dependencies from a local folder (offline behavior), while still downloading others from maven/sonatype (online behavior). With the current code, I can only either have all dependencies on a local folder or download all dependencies.
In my case, I only wanted to use my local shadows-core library and I didn't want to download all other dependencies to do that.

Based on comments on pull 1575, here are the changes:
- Crested a CompositeDependencyResolver that will try to resolve dependencies asking the resolvers in the order they were added.
- Modify TestRunner to use CompositeDependencyResolver.
- Checking local dependency folder at environment variable ROBOLECTRIC_DEPENDENCY_DIR a well as on system variable robolectric.dependency.dir

### CODE REVIEW

File: robolectric/src/main/java/org/robolectric/RobolectricTestRunner.java [hunk 1]
```diff
@@ -98,24 +100,24 @@ private void assureTestLifecycle(SdkEnvironment sdkEnvironment) {
...
+      dependencyResolver = new CompositeDependencyResolver(Arrays.asList(
+              new LocalDependencyResolver(dependencyDir == null?null:new File(dependencyDir)),
+              new CachedDependencyResolver(new MavenDependencyResolver(), cacheDir, CACHE_EXPIRE_TIME)));
     }
-
     return dependencyResolver;
   }
 
+  private String getLocalDependencyDir(){
+    String dependencyDir = System.getenv("ROBOLECTRIC_DEPENDENCY_DIR");
```
Comment (erd) [inline robolectric/src/main/java/org/robolectric/RobolectricTestRunner.java]: We don't currently expose any configuration via environment variables, so I'd prefer to do this in a separate commit (and potentially do it for all of robolectric's configuration).

File: robolectric/src/main/java/org/robolectric/internal/dependency/CompositeDependencyResolver.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package org.robolectric.internal.dependency;
+
+import java.net.URL;
+import java.util.List;
+
+public class CompositeDependencyResolver implements DependencyResolver {
+  private List<DependencyResolver> dependencyResolvers;
+
+  public CompositeDependencyResolver() {
```
Comment (erd) [inline robolectric/src/main/java/org/robolectric/internal/dependency/CompositeDependencyResolver.java]: Does a default constructor make sense here? Especially since you can't set the dependencyResolvers later. I'd vote to remote it.

File: robolectric/src/main/java/org/robolectric/internal/dependency/CompositeDependencyResolver.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    super();
+  }
+
+  public CompositeDependencyResolver(List<DependencyResolver> dependencyResolvers) {
+    this();
```
Comment (erd) [inline robolectric/src/main/java/org/robolectric/internal/dependency/CompositeDependencyResolver.java]: The call to `this()` isn't needed.

File: robolectric/src/main/java/org/robolectric/internal/dependency/CompositeDependencyResolver.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
+package org.robolectric.internal.dependency;
+
+import java.net.URL;
+import java.util.List;
+
+public class CompositeDependencyResolver implements DependencyResolver {
+  private List<DependencyResolver> dependencyResolvers;
```
Comment (erd) [inline robolectric/src/main/java/org/robolectric/internal/dependency/CompositeDependencyResolver.java]: Mark as final (if done with the comment below).

**Comment by fhsilva** [comment]
Hi Erich, I updated the commit with your suggestions.

**Comment by erd** [comment]
I merged this into my local tree and I'm seeing a ton of test failures that look similar to the Travis failures above. Could you take a look at those?

**Comment by erd** [comment]
Feel free to re-open this PR if you get a chance to address the test failures I mentioned above.
